### DISCUSSION TITLE
Does someone have a build sequence for making plugins?

### DISCUSSION THREAD

**Discussion by Marxsal** [TiddlyWiki/TiddlyWiki5#5500] [Q&A] [answered]
Sometimes you want to generate a plugin without having to build a wiki and then extract the plugin.

Is there a suggested sequence that could be inserted into tiddlywiki.info for making just the plugin?

Thanks

**Comment by clutterstack** [comment]
While waiting for an authoritative answer, and at the risk of misunderstanding your needs...would the `library` build target of the `pluginlibrary` edition do the job?

**Reply by Marxsal** [reply]
That sounds interesting. It's not mentioned on tiddlywiki.com. Do you have a link to it?

Offhand, I think making a library is more than I had in mind. I just want to generate a plugin without having to first generate a stand-alone file, and then navigate to the plugin, and then export it. 

Thanks!

**Reply by kookma** [reply]
Hi @Marxsal - do you mean to generate a plugin inside browser using a single html file? If not using tiddlywiki commands with nodejs you can create a plugin without needing to generate a new single html file!
By the way I am adding a feature to Tiddler Commander which creates plugins on the fly. Just filter few tiddlers and press a button. I hope to publish in March. 
In another thread I asked @Jermolene to advise how create another type of plugins like a presentation, a course note, an ebook, etc... If Jeremy decide on that I will ready to start to work on the Commander addon to generate a plugin or an e-book from a set of filtered tiddlers.

**Reply by Marxsal** [reply]
> Hi @Marxsal - do you mean to generate a plugin inside browser using a single html file? If not using tiddlywiki commands with nodejs you can create a plugin without needing to generate a new single html file!

Do you mean with Tinka? Or is there some other way that I've missed? In my case, I want to use a text editor, so I do need node. 

Thanks!

**Comment by Jermolene** [comment, marked as answer]
Hi @Marxsal try something like this:

```bash
tiddlywiki editions/prerelease/ --render "$:/plugins/tiddlywiki/codemirror" "[slugify[]addsuffix[.json]]" "text/plain" "$:/core/templates/json-tiddler"
```

The first argument is actually a filter, so you can easily save multiple plugins with a single command.

**Reply by Jermolene** [reply]
This version saves all the plugins in the wiki:

```
tiddlywiki editions/prerelease/ --render "[has[plugin-type]]" "[slugify[]addsuffix[.json]]" "text/plain" "$:/core/templates/json-tiddler"
```

**Reply by kookma** [reply]
> ```shell
> tiddlywiki editions/prerelease/ --render "$:/plugins/tiddlywiki/codemirror" "[slugify[]addsuffix[.json]]" "text/plain" "$:/core/templates/json-tiddler"
> ```

Hi @Jermolene  - what is "$:/plugins/tiddlywiki/codemirror" in the above command?

**Reply by Jermolene** [reply]
> Hi @Jermolene  - what is "$:/plugins/tiddlywiki/codemirror" in the above command?

It's the title of the plugin that you want to save.

**Reply by Marxsal** [reply]
> tiddlywiki editions/prerelease/ --render "$:/plugins/tiddlywiki/codemirror" "[slugify[]addsuffix[.json]]" "text/plain" "$:/core/templates/json-tiddler"
> ```
> 
> The first argument is actually a filter, so you can easily save multiple plugins with a single command.

Thanks Jeremy! That's what I was looking for. Now if I only had some place to store these little bits of information ...
